addons/mail/models/mail_channel.py

```diff
@@ -405,6 +439,49 @@ def _alias_get_error_message(self, message, message_dict, alias):
             return False
         return super(Channel, self)._alias_get_error_message(message, message_dict, alias)
 
+    def _notify_compute_recipients(self, message, msg_vals):
+        """ Override recipients computation as channel is not a standard
+        mail.thread document. Indeed there are no followers on a channel.
+        Instead of followers it has members that should be notified.
+
+        :param message: see ``MailThread._notify_compute_recipients()``;
+        :param msg_vals: see ``MailThread._notify_compute_recipients()``;
+
+        :return recipients: structured data holding recipients data. See
+          ``MailThread._notify_thread()`` for more details about its content
+          and use;
+        """
+        recipients_data = []
+        mailing_channels = self.filtered('email_send')
+        if mailing_channels:
+            msg_sudo = message.sudo()
+            email_from = tools.email_normalize(msg_vals.get('email_from') or msg_sudo.email_from)
+            author_id = msg_vals.get('author_id') or message.author_id.id
+            except_pids = [author_id] if author_id else []
+            sql_query = """
+                SELECT DISTINCT ON (partner.id) partner.id
+                  FROM res_partner partner
+             LEFT JOIN mail_channel_partner mcp on partner.id = mcp.partner_id
+             LEFT JOIN mail_channel channel on channel.id = mcp.channel_id
+             LEFT JOIN res_users users on partner.id = users.partner_id
+                 WHERE (users.notification_type != 'inbox' or users.id IS NULL)
+                       AND (partner.email != %s or partner.email IS NULL)
+                       AND channel.id = ANY(%s)
+                       AND partner.id != ANY(%s)"""
+            self.env.cr.execute(sql_query, (email_from, mailing_channels.ids, except_pids, ))
+            for partner_id in self._cr.fetchall():
+                # ocn_client: will add partners to recipient recipient_data. more ocn notifications. We neeed to filter them maybe
+                recipients_data.append({
+                    'id': partner_id[0],
+                    'share': True,
+                    'active': True,
+                    'notif': 'email',
+                    'type': 'channel_email',
+                    'groups': []
+                })
+
+        return {'partners': recipients_data, 'channels': []}
```

Comment:
Channel is still here in the final diff

Reply:
I made a final cleaning commit to remove it from result, cross-apps. It was planned but not yet done, so we agree on it :)